main-darwin.cpp

```diff
@@ -123,9 +123,7 @@ assert_handler (const jerry_api_object_t *function_obj_p __attr_unused___, /** <
                 const jerry_api_value_t args_p[], /** < function arguments */
                 const jerry_api_length_t args_cnt) /** < number of function arguments */
 {
-  if (args_cnt == 1
-      && args_p[0].type == JERRY_API_DATA_TYPE_BOOLEAN
-      && args_p[0].v_bool == true)
+  if (args_cnt == 1 &&  && jerry_api_value_to_boolean (&args_p[0]))
```

Comment:
The conversion hides the some kind of issues. For example, `assert ("abc".replace("abc", "$x$$5$0$00$" === "$x$5$0$00$"));` should be `assert ("abc".replace("abc", "$x$$5$0$00$") === "$x$5$0$00$");`, but assertion passes, and so, possible issue becomes hidden.

Please, see also https://github.com/Samsung/jerryscript/issues/711#issuecomment-155385250 and issue #716.